### ISSUE TITLE
[Bug] train_step()` retuned `None` outputs. Skipping training step.

### ISSUE THREAD

**Issue by DeScoutist** [coqui-ai/TTS#3166]
### Describe the bug

I have 90 sound files in .wav format. Script works, and training begins. 
The config file is next:
{
   "model": "glow_tts",
   "batch_size": 15,
   "eval_batch_size": 5,
   "num_loader_workers": 4,
   "num_eval_loader_workers": 4,
   "run_eval": true,
   "test_delay_epochs": -1,
   "epochs": 1000,
   "text_cleaner": "english_cleaners",
   "use_phonemes": false,
   "phoneme_language": "en-us",
   "phoneme_cache_path": "phoneme_cache",
   "print_step": 25,
   "print_eval": true,
   "mixed_precision": false,
   "output_path": "recipes/ljspeech/glow_tts/",
   "test_sentences": ["Test this sentence.", "This test sentence.", "Sentence this test."],
   "datasets":[{"formatter": "nzothvoice", "meta_file_train":"metadata.txt", "path": "recipes/ljspeech/MyDataset/"}]

But on the first Epoch it gives me that error "train_step()` retuned `None` outputs. Skipping training step." 6 times. On the second epoch - 2-4 times and then it always gives me: 
--> STEP: 0
     | > loss: 2.829817056655884  (2.829817056655884)
     | > log_mle: 0.551344633102417  (0.551344633102417)
     | > loss_dur: 2.278472423553467  (2.278472423553467)
(data varies from time to time)

How should I debug that? Never found any mention of this case here. :/

(also sometime "warning: audio amplitude out of range, auto clipped." shows up but script continues anyway)

### To Reproduce

I just turn on the script train_tts.py with 90 audio files in .wav and config given above;

### Environment

```shell
{
    "CUDA": {
        "GPU": [],
        "available": false,
        "version": "12.1"
    },
    "Packages": {
        "PyTorch_debug": false,
        "PyTorch_version": "2.1.0+cu121",
        "TTS": "0.20.0",
        "numpy": "1.26.1"
    },
    "System": {
        "OS": "Linux",
        "architecture": [
            "64bit",
            "ELF"
        ],
        "processor": "x86_64",
        "python": "3.10.12",
        "version": "#1 SMP Fri Jan 27 02:56:13 UTC 2023"
    }
}
```

### Additional context

(I try to go AI on CPU)

**Comment by DeScoutist** [comment]
It just shown me the next:

> EPOCH: 70/1000
 --> recipes/ljspeech/glow_tts/run-November-08-2023_08+16AM-0000000

 > TRAINING (2023-11-08 08:40:24) 

   --> TIME: 2023-11-08 08:40:40 -- STEP: 5/6 -- GLOBAL_STEP: 425
     | > loss: 2.9993789196014404  (3.2876376152038573)
     | > log_mle: 0.20223742723464966  (0.19471521377563478)
     | > loss_dur: 2.7971415519714355  (3.092922401428223)
     | > grad_norm: tensor(4.4177)  (tensor(4.8612))
     | > current_lr: 1.7500000000000002e-05
     | > step_time: 3.4633  (2.2653998851776125)
     | > loader_time: 0.0024  (0.0017766952514648438)

So, the steps can eventually work. But how to understand that? My AI can't learn from epochs with no steps, right?

**Comment by erogol** [comment]
We need a way to reproduce the error. Can't help you without.

**Comment by DeScoutist** [comment]
I have no idea how to recreate it on anyone's else machine, I just started the script after all the preparation steps done with my audio and changing all cuda references in script to the cpu, where possible. :/

I just got that it is defined in config.json : "print_step": 25, which means that only each 25th global step get recorded, so it might be not skipped, because "train_step() returned None outputs" only called 10 first times. 

Am I right that if "train_step()" error is only in the first 10 steps of the code - it means it never get called again after?

Also, I tried to launch TensorBoard but it gives me no info as it sees no data in the folder of the run. Should TensorBoard's files be created immediately after run started?

**Comment by erogol** [comment]
Best, you can crate a sample project on colab and share the link here or DM me.

**Comment by DeScoutist** [comment]
> Best, you can crate a sample project on colab and share the link here or DM me.

I'm sorry, kinda new to all those python thingies... Do you mean that I need to copy my whole project to Google Collab, or should I somehow provide you something specified?

I'm asking because I don't see any button I can put whole project in... :/

I can share my dataset I try to use. I think it might be something with mel's or maybe other sound things, I didn't fixed in some proper way because of experience. ( https://drive.google.com/drive/folders/1Ezj0EeK_dnePmzWa31kwS4TgIpaQUHrp?usp=drive_link )

!!! Here is the way to recreate this bug using downloaded from google Disk dataset !!!

1) Put the config.json file into /root/TTS/tts/configs (in case you use Docker as I do) OR /TTS/tts/configs if you not;
2) Unzip MyDataset into /root/recipes/ljspeech folder (in case you use Docker) OR /recipes/ljspeech if you not;
3) Run the "python /root/TTS/bin/train_tts.py --config_path /root/TTS/tts/configs/config.json " (I run this in my Docker container)

And then you will have 6 "train_step() retuned None outputs" in a row;

**Comment by DeScoutist** [comment]
What also concerns me - is that I can't receive any TensorBoard images from my folder, because there are no suitable files in it, as TensorBoard's 6006 states. :/

**Comment by DeScoutist** [comment]
I got the next soundfile as a result of 5 hour training (on CPU, it is said that it should be much less effective):  https://drive.google.com/drive/u/0/folders/1Ezj0EeK_dnePmzWa31kwS4TgIpaQUHrp

(files "in.wav" and "out(5 hour training on CPU).wav")

Is that right that this thing works?.. How to understand if I am being effective in training or I am tricking myself.
